### PR TITLE
Add the user test program in HDFFV-9174 for committed types.

### PR DESCRIPTION
**Pull request by vchoi-hdfgroup**

### CODE REVIEW

**Review by mattjala — APPROVED** [review]

File: test/dtypes.c [hunk 1]
```diff
@@ -7567,6 +7567,177 @@ test_named_indirect_reopen(hid_t fapl)
...
+ *
+ * Purpose: Tests that a named compound datatype that refers to a named
+ *          string datatype can be reopened indirectly through H5Dget_type,
+ *          and shows the correct H5Tcommitted() state, including after the
+ *          file has been closed and reopened.
+ *
+ * Return:  Success:    0
+ *          Failure:    number of errors
+ *
+ * Programmer:  Mark Hodson
```
Comment (derobins) [inline test/dtypes.c]: We no longer include "Programmer:" lines

File: test/dtypes.c [hunk 2]
```diff
@@ -7567,6 +7567,177 @@ test_named_indirect_reopen(hid_t fapl)
...
+test_named_indirect_reopen_file(hid_t fapl)
+{
+    hid_t          file             = H5I_INVALID_HID;
+    hid_t          space            = H5I_INVALID_HID;
+    hid_t          cmptype          = H5I_INVALID_HID;
+    hid_t          reopened_cmptype = H5I_INVALID_HID;
+    hid_t          strtype          = H5I_INVALID_HID;
+    hid_t          reopened_strtype = H5I_INVALID_HID;
+    hid_t          dset             = H5I_INVALID_HID;
+    static hsize_t dims[1]          = {3};
```
Comment (derobins) [inline test/dtypes.c]: There's no reason for this to be static

File: test/dtypes.c [hunk 3]
```diff
@@ -7567,6 +7567,177 @@ test_named_indirect_reopen(hid_t fapl)
...
+    if ((strtype = H5Tcopy(H5T_C_S1)) < 0)
+        TEST_ERROR;
+    if (H5Tset_size(strtype, H5T_VARIABLE) < 0)
+        TEST_ERROR;
+
+    /* Get size of string type */
+    if ((strtype_size = H5Tget_size(strtype)) == 0)
+        TEST_ERROR;
+
+    /* Commit compound type and verify the size doesn't change */
```
Comment (derobins) [inline test/dtypes.c]: This commits the string type, not the compound type

File: test/dtypes.c (hunk 1 above)
Comment (vchoi-hdfgroup) [reply test/dtypes.c]: Removed.

File: test/dtypes.c (hunk 2 above)
Comment (vchoi-hdfgroup) [reply test/dtypes.c]: Corrected.

File: test/dtypes.c (hunk 3 above)
Comment (vchoi-hdfgroup) [reply test/dtypes.c]: Corrected.

**Review by derobins — APPROVED** [review]

**Pull request merged**